### ISSUE TITLE
Support more Response-types

### ISSUE THREAD

**Issue by ikitommi** [metosin/pohjavirta#5]
* `File`, `InputStream` at least

**Comment by ohpauleez** [comment]
As always, love to see this stuff.  Current version of "Pedestal2" has many similarities with the approach here.

You might consider changing from `CompletableFuture` to `CompletionStage` in response.clj.  (More than happy to elaborate here, but I think you'll get the main idea).

**Comment by ikitommi** [comment]
Good catch, thanks! Curious about the "Pedestal2", didn't find anything on the pedestal repo, any more info available? We are in process of pulling a set of libraries into a new opinionated developer-friendly stack, will end up here: https://github.com/metosin/talvi

Logo-driven development ;)

**Comment by malchmih** [comment]
Same here, would love for this project to take off!

**Comment by kanwei** [comment]
It definitely seems like implementing this ticket would allow adoption by folks like me.
